[BUG]: flake8 checks untracked files in repository

### Description of the bug

Executing a git commit ends with error messages from `flake8` for untracked files, which I can reproduce by executing `python -m flake8` in the repo (see Steps To Reproduce for command line execution and output).

Note that for me, `source tools/pre-commit.sh` threw the following error message `.git/hooks/pre-commit: 1: source: not found` when trying to commit, and the way I worked around this (after googling) was to switch to `. tools/pre-commit.sh`. `source tools/pre-commit.sh` works from the command line however, but exits the shell connection at the end (another mystery to me).

In theory, what I would expect is flake8 to work on the changed files that `git status -s` would tell me about, and if I were to run `python -m flake8 $(git status -s | grep -Ev '\?\?' | cut -c 4- )`, it does seem to act on the 3 files that were changed. Notice, however that this time, `flake8` throws an error for `armory/utils/config_schema.json`, which it did not do for the previous commit attempt. Again, it is unclear to me what `flake8` is targeting with the configuration set up in `.flake8` and how it works within the context of the overall commit and prehook process.



### Steps To Reproduce

```
(armory_repo) (base) paul.park@noether:~/repositories/armory$ git status
On branch pp/tide
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   armory/metrics/task.py
	modified:   armory/scenarios/main.py
	modified:   armory/utils/config_schema.json

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	dev_notes/
	tide_integration_test.ipynb

(armory_repo) (base) paul.park@noether:~/repositories/armory$ cat .git/hooks/pre-commit
. tools/pre-commit.sh

(armory_repo) (base) paul.park@noether:~/repositories/armory$ git commit -m "tested adding new metric task to scenario; tested importing tidecv and converting armory output into TIDE input within new task definition for a single image - need to generalize for a list of images; need to determine what to return and in what format to be compatible with post-processing that seems to happen after the call is done"
./.ipynb_checkpoints/metrics-checkpoint.py:5:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:37:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:142:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:164:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./.ipynb_checkpoints/metrics-checkpoint.py:206:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./.ipynb_checkpoints/metrics-checkpoint.py:207:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:210:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:290:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:291:9: E303 too many blank lines (2)
./.ipynb_checkpoints/metrics-checkpoint.py:293:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:295:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:300:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:513:39: W292 no newline at end of file
./dev_notes/metrics.py:165:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./dev_notes/metrics.py:207:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:5:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:37:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:142:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:164:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:206:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:207:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:210:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:290:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:291:9: E303 too many blank lines (2)
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:293:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:295:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:300:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:513:39: W292 no newline at end of file

(armory_repo) (base) paul.park@noether:~/repositories/armory$ python -m flake8
./.ipynb_checkpoints/metrics-checkpoint.py:5:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:37:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:142:1: E302 expected 2 blank lines, found 1
./.ipynb_checkpoints/metrics-checkpoint.py:164:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./.ipynb_checkpoints/metrics-checkpoint.py:206:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./.ipynb_checkpoints/metrics-checkpoint.py:207:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:210:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:290:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:291:9: E303 too many blank lines (2)
./.ipynb_checkpoints/metrics-checkpoint.py:293:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:295:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:300:1: W293 blank line contains whitespace
./.ipynb_checkpoints/metrics-checkpoint.py:513:39: W292 no newline at end of file
./dev_notes/metrics.py:165:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./dev_notes/metrics.py:207:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:5:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:37:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:142:1: E302 expected 2 blank lines, found 1
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:164:5: F841 local variable 'RECALL_POINTS' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:206:5: F841 local variable 'average_precisions_by_class' is assigned to but never used
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:207:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:210:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:290:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:291:9: E303 too many blank lines (2)
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:293:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:295:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:300:1: W293 blank line contains whitespace
./dev_notes/.ipynb_checkpoints/metrics-checkpoint.py:513:39: W292 no newline at end of file

(armory_repo) (base) paul.park@noether:~/repositories/armory$ git status -s | grep -Ev '\?\?' | cut -c 4-
armory/metrics/task.py
armory/scenarios/main.py
armory/utils/config_schema.json

(armory_repo) (base) paul.park@noether:~/repositories/armory$ python -m flake8 $(git status -s | grep -Ev '\?\?' | cut -c 4- )
armory/utils/config_schema.json:112:25: F821 undefined name 'null'
```

### Additional Information

_No response_

--- a/docs/style.md
+++ b/docs/style.md
@@ -28,8 +28,12 @@ Our repo-specific Flake8 configuration is detailed in `.flake8`.
 The above tools can be ran automatically, prior to each commit, by sourcing `tools/pre-commit.sh` in `.git/hooks/pre-commit`. See the example below:
 
 ```bash
-echo "source tools/pre-commit.sh" > .git/hooks/pre-commit
+echo "./tools/pre-commit.sh" > .git/hooks/pre-commit
 chmod 755 ".git/hooks/pre-commit"
+```
+NOTE: if there are shell errors when the pre-commit hook is launched, you may need to modify it to your shell of choice:
+```bash
+echo "<shell> tools/pre-commit.sh" > .git/hooks/pre-commit
 ```
 
 There is an `EARLY_EXIT` variable in `tools/pre-commit.sh` that can be modified that causes the script to exit on the first failure.

